### ISSUE TITLE
Edge: allow NvDA to navigate Epub books in Edge in Windows 10 Creators Update

### ISSUE THREAD

**Issue by josephsl** [nvaccess/nvda#6925]
Hi,

In Windows 10 Creators Update, you can use Microsoft Edge to read Epub books, similar to what Edge can do now to read PDF's. I propose adding support for Epub book reading in Edge (a good test case is an Epub version of NVDA Basic Training book).

Thanks.

**Comment by dkager** [comment]
Very interested in this. Any ideas on what it would take?

**Comment by jcsteh** [comment]
CC @MichaelDCurran.

**Comment by nvdaes** [comment]
Replying to @josephsl 
> (a good test case is an Epub version of NVDA Basic Training 

I think it should be use an ePUB 3.0 file, for instance the proposed in epubtest.org to evaluate accessibility of reading systems.
Unfortunately, that website doesn't seems to mantain permanent links of evaluations, so hope you can find Edge evaluation with JAWS.
I'm evaluating Edge and other ePUB reading systems with NVDA performing the proposed tests, and I think that the book containing these test is useful to provide specific aspects to be fixed. For now I see lack, at least, in add and review notes, and find info about current position. The search feature works great. I don't have Edge here (on this computer), but I'm searching information to continue with evaluations and have found this issue.
Now, Edge is evaluated with JAWS at
http://epubtest.org/testsuite/accessibility/

Perhaps I will posts results for Edge with NVDA evaluation the next weekend

**Comment by nvdaes** [comment]
Here are results of Edge Evaluation with the ePUB Accessibility Test (fundamental, i.e., all items are required), not the optional tests.
In short, all test relative to file and Non Visual Reading pass, these last with comments. Some navigation and annotation tests fail. Comments and an NVDA log fragment are provided in the attached file.
Thanks
[Edge.txt](https://github.com/nvaccess/nvda/files/1460196/Edge.txt)

**Comment by nvdaes** [comment]
Hi, I'm now part of a mailing list of people who performs test posted on epubtest.org, about reading systems in combination with assistive technologies. Yesterday it was sent a document with suggestion about Edge on the last Windows 10 1803, and I evaluated Edge with NVDA 2018.1.1 on Win10 1803, and also with a master version, which seems to work better than 2018.1.1, so I will evaluate Edge on the epubtest.org website when NVDA 2018.2 is released.
According with my evaluation performing the proposed tests, all seems to work right, but the main problem is that often words are splitted in multiple lines, for instance:
set
tings...
This has impact on reading, and it doesn't seem to happen with Narrator, but I prefer NVDA.
Notes can also be added using the Edge cursor (pressing f7 to activate this feature), and switching to focus mode. Anyway, sometimes it's not easy to move this cursor by lines, and right arrow needs to be used, or even tab and f6 to focus the Window and selecting text. Then the context menu can be used to add a note, and shortcut keys are available to review notes, table of contents, bookmarks, hub, etc.
For me, except this comment, now Edge with NVDA masters versions, on Windows 10 1803, is the only reading system which could be evaluated, with comments, as 100% accessible in basic and non visual tests.
Hope the problem of lines can be fixed.

**Comment by nvdaes** [comment]
Testing again, I see that the number of characters in lines deppend on zoom, that is, they can be changed with control+plus and control+minus. So, I think that this may be fixed if screen layout could be disabled on Edge.

Thanks

**Comment by derekriemer** [comment]
Wow, that's cool. It would be best if we could select in browse mode then get the notes and highlights.

**Comment by nvdaes** [comment]
Evaluation with NVDA 2018.1.1 has been published in epubtest.org.
After that, I think this issue maybe invalid (too wide or unspecific) and splitted in different ones, since ePUB navigation in Edge with NVDA is possible, with precisse problems which may affect other kind of web contents:
http://epubtest.org/evaluation/467/
Thanks

**Comment by Adriani90** [comment]
@nvdaes how is this in Edge Chromoum 81 or newer? I suggest to close this if it works as expected in Edge Chromium because it will be distributed to most users anyway through the Windows 10 20 H1 update.

**Comment by nvdaes** [comment]
Thanks, I wantet to suggest to @josephsl to close it and I delayed it unfortunately. As far as I know, Edge has removed support for epub and it is not recommended by @Daisy now. The most accessible open source app based on Chronium that I use with NVDA as my epub reader is Thorium.
cc: @danielweck

**Comment by Brian1Gaff** [comment]
Does this feature depend on operating system, as there is now an edge for 
Windows 7, would that work?
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users

**Comment by Adriani90** [comment]
yes, that would work.

Thanks @nvdaes for reporting back. I will closing this isue for now. We can reopen this discussion if needed.

**Comment by nvdaes** [comment]
Thank you @Adriani90 .
For reference, here is [Thorium reader website](https://www.edrlab.org/software/thorium-reader/).
Here are [useful keystrokes for Thorium](https://github.com/readium/readium-desktop/wiki/Keyboard-shortcuts).
Here the [Roundup about epub reading system](https://inclusivepublishing.org/rs-accessibility/). Note that Edge is not mentioned in the latest version of the document as an application with capabilities to read epub itself, though an extension can be installed for this.
